### PR TITLE
feat(@vtmn/react): add `VtmnSearch` component

### PR DESCRIPTION
**Pull request by 0x2A-git**
## Changes description

This P.R adds the search component along with its showcase

## Context

Related issue #821 

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on related showcases.
- [x] If it includes design changes, please ask for a review with a core team designer.

## Does this introduce a breaking change?

- No

### CODE REVIEW

File: packages/sources/react/src/index.ts [hunk 1]
```diff
@@ -31,6 +31,8 @@ export { VtmnProgressbar } from './components/indicators/VtmnProgressbar';
 export type { VtmnProgressbarProps } from './components/indicators/VtmnProgressbar';
 
 // Navigation
+export { VtmnSearch } from './components/navigation/search/VtmnSearch';
+export type { VtmnSearchProps } from './components/navigation/search/VtmnSearch';
```
Comment (lauthieb) [inline packages/sources/react/src/index.ts:35]: Can you please rename the "search" folder to "VtmnSearch" as we did with other components please? :)

**Review by lauthieb — CHANGES_REQUESTED** [review]
Hi @0x2A-git thanks for your proposal! Just a few change for the moment before going deeper to your PR :)

**Comment by lauthieb** [comment]
I convert to draft, can you please fix conflicts @0x2A-git ? :)

**Comment by lauthieb** [comment]
I convert to draft, can you please fix conflicts @0x2A-git ? :)

**Comment by 0x2A-git** [comment]
> I convert to draft, can you please fix conflicts @0x2A-git ? :)

It's done :), thanks for the report I didn't notice it :woozy_face:

**Review by lauthieb — APPROVED** [review]
Thanks a lot! :)

**Pull request merged**